packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -577,6 +572,21 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       };
   }
 
+  // Initialize the text field with the label associated to the entry
+  // whose value matches the initial selection.
+  // Empty the text field when no entry matches the initial selection.
+  void matchInitialSelection() {
+    final int index = filteredEntries.indexWhere((DropdownMenuEntry<T> entry) => entry.value == widget.initialSelection);
+    if (index != -1) {
+      _localTextEditingController?.value = TextEditingValue(
+        text: filteredEntries[index].label,
+        selection: TextSelection.collapsed(offset: filteredEntries[index].label.length),
+      );
```

Comment:
Tiny nit—we could reduce the number of times the list is accessed:

```suggestion
      final String label = filteredEntries[index].label;
      _localTextEditingController?.value = TextEditingValue(
        text: label,
        selection: TextSelection.collapsed(offset: label.length),
      );
```

Reply:
Nice! Typical case where I moved the existing code without reading it very closely :grin: